lib/authorizer/oauth2.js

```diff
@@ -58,7 +59,46 @@ module.exports = {
      * @param {AuthHandlerInterface~authPreHookCallback} done -
      */
     pre: function (auth, done) {
-        done(null, Boolean(auth.get('accessToken')));
+        const authId = auth.get('authId'),
+            refreshTokenHelper = auth.get('refreshTokenHelper');
+
+        // Go into the refresh token flow ONLY IF we have all necessary information present
+        if (!(authId && refreshTokenHelper && _.isFunction(refreshTokenHelper.refreshToken))) {
+            return done(null, Boolean(auth.get('accessToken')));
+        }
+
+        let refreshTokenObject = refreshTokenHelper.refreshToken(authId),
```

Comment:
Why not return the promise itself?

```suggestion
        let promise = refreshTokenHelper.refreshToken(authId),
```